arelle/plugin/validate/EDINET/rules/gfm.py

```diff
@@ -1184,6 +1184,35 @@ def rule_gfm_1_7_3(
             )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_7_4(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.7.4] A calculation link can not contain directed cycles.
+    """
+    cycleObjects = set()
+    fromRelationships = val.modelXbrl.relationshipSet(tuple(LinkbaseType.CALCULATION.getArcroles())).fromModelObjects()
```

Comment:
Cycles are usually defined per arcrole (`cyclesAllowed`), but this seems to detect cycles across arcroles.  Is that deliberate?

Or is the rule just intending to specify `cyclesAllowed` for the XBRL 2.1 calc arcrole, which otherwise doesn't specify a restriction (but was considered an oversight and fixed in calc 1.1)?  If so, I wonder if it would be simpler to adjust `XbrlValidate` to allow plugin-specific `standardArcroleCyclesAllowed`.